### PR TITLE
Add support for filtering a watch action based on label selectors.

### LINKED ISSUE #7: IKubernetesClient.Watch<TResource> should support specifying label selectors
Presently, `IKubernetesClient` supports passing a label selector either as a string, or as a collection of `ILabelSelector`, to `List<TResource>`.

https://github.com/buehler/dotnet-kubernetes-client/blob/f65cb14b2c5eded11aa33c0601ebc443d41f58cf/src/DotnetKubernetesClient/IKubernetesClient.cs#L90-L93

https://github.com/buehler/dotnet-kubernetes-client/blob/f65cb14b2c5eded11aa33c0601ebc443d41f58cf/src/DotnetKubernetesClient/IKubernetesClient.cs#L76-L79

In contrast to that, however, `Watch<TResource>` only allows specifying the resource kind (through the type parameter) and will watch all instances of that resource, either in the given namespace or globally.

https://github.com/buehler/dotnet-kubernetes-client/blob/f65cb14b2c5eded11aa33c0601ebc443d41f58cf/src/DotnetKubernetesClient/IKubernetesClient.cs#L186-L193

___

The associated method invocation in the IKubernetes instance has arguments enough to support this without massive code change (can't inline link because the source file is too large for GH 🤷‍♀️ ):

```csharp
public async Task<HttpOperationResponse<object>> ListClusterCustomObjectWithHttpMessagesAsync(
	string group,
	string version,
	string plural,
	string continueParameter = default(string),
	string fieldSelector = default(string),
	string labelSelector = default(string),
	int? limit = default(int?),
	string resourceVersion = default(string),
	int? timeoutSeconds = default(int?),
	bool? watch = default(bool?),
	string pretty = default(string),
	Dictionary<string,
	List<string>> customHeaders = null,
	CancellationToken cancellationToken = default(CancellationToken))
```

My thought is that it should be relatively simple to add `string labelSelector` to `Watch<TResource>` so it can be passed into the above method, and then add an overload that supports the `ILabelSelector` type as well.

I'll throw that together in the morning and (assuming it works) send a PR for review.

### PR DESCRIPTION
**Pull request by erin-allison**
This would resolve #7.

Changes were tested for namespace and cluster-scoped resources on Kubernetes v1.19.5

### CODE REVIEW

File: src/DotnetKubernetesClient/IKubernetesClient.cs [hunk 1]
```diff
@@ -181,6 +181,7 @@ Task Delete<TResource>(string name, string? @namespace = null)
         /// If the namespace is omitted, all resources on the cluster are watched.
         /// </param>
         /// <param name="cancellationToken">Cancellation-Token.</param>
+        /// <param name="labelSelector">A string, representing an optional label selector for filtering watched objects.</param>
```
Comment (buehler) [inline src/DotnetKubernetesClient/IKubernetesClient.cs:184]: Not to be picky, but here, the documentation is not quite correct.
```suggestion
        /// <param name="labelSelector">A list of label selectors to apply to the watcher.</param>
```

**Review by buehler — CHANGES_REQUESTED** [review]
Good idea to add this!
Just one little comment regarding the documentation.

Thanks for the work

File: src/DotnetKubernetesClient/IKubernetesClient.cs (hunk 1 above)
Comment (erin-allison) [reply src/DotnetKubernetesClient/IKubernetesClient.cs:184]: Oh, oops; not only that, but the param name is incorrect... it should be `labelSelectors`, to match `List<TResource`

Comment (erin-allison) [reply src/DotnetKubernetesClient/IKubernetesClient.cs]: Fixed.

**Review by buehler — APPROVED** [review]

**Pull request merged**